gems/pending/openstack/openstack_event_monitor.rb

```diff
@@ -67,19 +67,30 @@ def each
   cache_with_timeout(:event_monitor_cache) { Hash.new }
 
   def self.event_monitor_class(options)
+    event_monitor_class = case options[:events_monitor]
+                          when :ceilometer
+                            OpenstackCeilometerEventMonitor
+                          when :amqp
+                            OpenstackRabbitEventMonitor
+                          else
+                            OpenstackNullEventMonitor
+                          end
+
     key = event_monitor_key(options)
-    event_monitor_class_cache[key] ||= begin
-      detected_event_monitor = subclasses.detect do |event_monitor|
-        begin
-          event_monitor.available?(options)
-        rescue => e
-          $log.warn("MIQ(#{self}.#{__method__}) Error occured testing #{event_monitor}
-                     for #{options[:hostname]}. Trying other AMQP clients.  #{e.message}")
-          false
-        end
-      end
-      detected_event_monitor || OpenstackNullEventMonitor
+    event_monitor_class_cache[key] ||= available_event_monitor(event_monitor_class, options)
+  end
+
+  def self.available_event_monitor(event_monitor_class, options)
+    monitor_available = begin
+      event_monitor_class.available?(options)
+    rescue => e
+      $log.warn("MIQ(#{self}.#{__method__}) Error occured testing #{event_monitor}
+                 for #{options[:hostname]}. Trying other AMQP clients.  #{e.message}")
```

Comment:
Need to update this error message.  Before we tried multiple event monitors for AMQP.  Now, we're only trying the one that's specified by the user in the UI.

If that event monitor doesn't work, then we immediately fall back to `NullEventMonitor`.

I suggest:

```ruby
  $log.warn("MIQ(#{self}.#{__method__}) Error occurred testing #{event_monitor}
             for #{options[:hostname]}. Event collection will be disabled for 
             #{options[:hostname]}  #{e.message}")
```

Eventually, we should revisit this and bubble this problem up to the UI somehow.  Not for this PR, but need to think about how to do that here.